From 33f032bfb745bfd78133f28f340a6063bf40daef Mon Sep 17 00:00:00 2001
From: Matyas Selmeci <matyas@cs.wisc.edu>
Date: Tue, 25 Nov 2014 21:08:35 -0600
Subject: [PATCH 25/26] DEROctetString new exception (bc1.47)

The DEROctetString constructor can now throw an IOException if the
argument it gets could not be converted into the proper format. I catch
the exception and ignore it -- the code in which this happens adds
'm_policy' to a vector, so if m_policy is bogus then it will just be
omitted from the vector.
---
 src/org/glite/security/util/proxy/ProxyPolicy.java | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/org/glite/security/util/proxy/ProxyPolicy.java b/src/org/glite/security/util/proxy/ProxyPolicy.java
index 166c002..f740c55 100644
--- a/src/org/glite/security/util/proxy/ProxyPolicy.java
+++ b/src/org/glite/security/util/proxy/ProxyPolicy.java
@@ -18,6 +18,8 @@ limitations under the License.
 
 package org.glite.security.util.proxy;
 
+import java.io.IOException;
+
 import org.bouncycastle.asn1.ASN1Object;
 import org.bouncycastle.asn1.ASN1EncodableVector;
 import org.bouncycastle.asn1.ASN1OctetString;
@@ -142,7 +144,11 @@ public class ProxyPolicy extends ASN1Object {
         ASN1EncodableVector v = new ASN1EncodableVector();
         v.add(new ASN1ObjectIdentifier(m_oid));
         if (m_policy != null) {
-            v.add(new DEROctetString(m_policy));
+            try {
+                v.add(new DEROctetString(m_policy));
+            } catch (IOException e) {
+                // nothing here
+            }
         }
 
         return new DERSequence(v);
-- 
2.0.1

